clients/fides-js/src/lib/consent-value.ts

```diff
@@ -28,6 +28,9 @@ export const resolveConsentValue = (
   if (notice.current_preference) {
     return transformUserPreferenceToBoolean(notice.current_preference);
   }
+  if (notice.consent_mechanism === ConsentMechanism.NOTICE_ONLY) {
```

Comment:
this is only helpful if the notice is in an invalid state of `notice-only` but `has_gpc_flag=true`. but I added it to be safe!

Reply:
actually, looking at this again, my earlier comment isn't _quite_ accurate—the logic is

1. return user's current preference if they have one
2. for notice-only, return true
3. if gpc is enabled and applicable, return false
4. return the notice's default preference 

for notice only, normally the 4th step would kick in and we'd be okay. but if it's an invalid state of notice-only + `has_gpc_flag=true`, then step 3 would kick in and opt us out of the notice-only notice. so adding step 2 helps prevent that and also returns a lil earlier for notice-only which is nice. I think it's a case where the series of if statements explains it better than a comment (the comment above applies more to why I added it, but I think the end state sense by itself 👍 )